### PR TITLE
[11.0][ADD] Documentation to authenticate as Odoo user via JSON-RPC

### LINKED ISSUE #12: Support for authentication
Unless I overshot snippets of code in base_rest, it looks like the module does not allow to authenticate to private services in a RESTful way. I guess most of the APIs written with this framework will only be available to authenticated users, so it would be great to add support for authentication with a standard request shared among services.

By the way, how do you manage authentication with the current module? An RPC call to login and get the session?

### PR DESCRIPTION
**Pull request by warp10**
Closes #12

### CODE REVIEW

File: base_rest/readme/USAGE.rst [hunk 1]
```diff
@@ -114,4 +114,31 @@ to access to your services
...
 The HTTP GET 'http://my_odoo/my_services_api/ping' will be dispatched to the
-method ``PingService.search``
\ No newline at end of file
+method ``PingService.search``
+
+Authentication
+~~~~~~~~~~~~~~
+
+The authentication to use services must be specified into the controller by the
+key ``_default_auth``. Then you can use any authentication mechanism supported
```
Comment (lmignon) [inline base_rest/readme/USAGE.rst]:
```suggestion
key ``_default_auth``. 


.. code-block:: python

    from odoo.addons.base_rest.controllers import main

    class MyRestController(main.RestController):
        _root_path = '/rest_api/'
        _collection_name = my_module.services
        _default_auth  = 'user'

Then you can use any authentication mechanism supported
```

**Review by lmignon — CHANGES_REQUESTED** [review]
@warp10 Thank you for your proposal. I propose to add a code block to illustrate where to add the `_default_auth` key.

**Review by dreispt — APPROVED** [review]

**Comment by dreispt** [comment]
@lmignon This is an important piece of documentation.
Can it be merged and forward ported?

**Comment by lmignon** [comment]
@dreispt I've to fix the travis....

**Pull request closed without merging**